### ISSUE TITLE
Error: Client index -1 is invalid (ManageFighterThink)

### ISSUE THREAD

**Issue by CookiesIO** [Starblaster64/Vs-Saxton-Hale-2#47]
This happens when there's an engineer.

```
L 12/10/2016 - 15:02:49: [SM] Exception reported: Client index -1 is invalid
L 12/10/2016 - 15:02:49: [SM] Blaming: customPlugins/vsh2.smx
L 12/10/2016 - 15:02:49: [SM] Call stack trace:
L 12/10/2016 - 15:02:49: [SM]   [0] GetClientTeam
L 12/10/2016 - 15:02:49: [SM]   [1] Line 1900, modules/handler.sp::ManageFighterThink
L 12/10/2016 - 15:02:49: [SM]   [2] Line 564, vsh2.sp::Timer_PlayerThink
```

[The enemy check on this line](https://github.com/Starblaster64/Vs-Saxton-Hale-2/blob/51b22e8029e936954dd29babd63a41ff9418f359/addons/sourcemod/scripting/modules/handler.sp#L1900) should at least be `enemy > 0` and would be better off as `enemy > 0 and enemy <= MaxClients `.

**Comment by assyrianic** [comment]
if only unsigned datatypes...

**Comment by Starblaster64** [comment]
@assyrianic FYI, instead of manually closing all these issues, you can close them automatically when you commit a change.
E.g. When you fixed this issue, you could have put in the *description*: "Fixes #47". Replace #47 with whatever issue you're closing with that commit. You can use other keywords like Resolves, closes etc. if you want, there's a list somewhere if you search for it.

It makes it much easier to find what commit addresses what issues when you do it this way as well.
